Update exclusion rules when the URL is changed by history.pushState/popState or its hash changes

mrmr1993 opened the pull request:
This PR fixes #1578, using `chrome.webRequest` to check exclusion rules for frames that changed URL without a page load.

Side effects:
* we need the `webRequest` permission
* introduced `minimum_chrome_version: 41` so we know we can message frames individually
  - current version is 42
* event listeners are always registered, even when Vimium is disabled
  - justification in 6446cf0, the relevant commit

smblott-github commented on manifest.json:
```diff
@@ -2,6 +2,7 @@
   "manifest_version": 2,
   "name": "Vimium",
   "version": "1.49",
+  "minimum_chrome_version": "41",
```
@mrmr1993.  Thanks for this PR.  I'll take a proper look soon.

What aspect of the PR requires 41?  Is 40 not a bit too recent to be excluding?  It was just released in January (if I have it right).

mrmr1993 replied:
`chrome.tabs.sendMessage` only supports messaging individual frames from Chrome 41. #1575 needs 41+ for the same reason.

> Is 40 not a bit too recent to be excluding?

According to [here](http://gs.statcounter.com/#desktop-browser_version-ww-monthly-201504-201504-bar), Chrome 41/42 have 77.8% of Chrome's market total, Chrome 40 has just 2.8%. Still, I'm happy to remove the minimum version and add a check around the version-sensitive code, if that would make you less uncomfortable.

smblott-github commented:
@mrmr1993.

> chrome.tabs.sendMessage only supports messaging individual frames from Chrome 41. #1575 needs 41+ for the same reason.

Thanks for pointing this out.

My sense is that requiring 41 is leaving too many users out.  I'll rework #1575.  I'm reviewing #1573 and #1575 today, and hope/expect to merge them.  If it's ok with you, I'll then review, rework and probably merge this.

(Separately, whatever Chrome version which introduced `chrome.sessions` is now about 18 months old.  It's probably safe enough to take out the conditional code there now.)

mrmr1993 commented:
> requiring 41 is leaving too many users out.

Both this and your #1575 will need a rewrite to avoid errorring on Chromes <41, then. I'll try to do this after my exam today, and #1575 should be as simple as dropping `{frameId: 0}` and checking that `window == window.top` at the receiving end instead.

> If it's ok with you

It is.

> (Separately, whatever Chrome version which introduced `chrome.sessions` is now about 18 months old. It's probably safe enough to take out the conditional code there now.)

Since we require `chrome.sessions` &mdash; which Chrome 36 doesn't have, &mdash; I believe Chrome 36 users wouldn't have been able to install Vimium 1.49 (released 5 months after Chrome 36 landed) regardless. This means that the conditional was useless from the moment we added the `sessions` permission.

I held off from mentioning this before, but Chrome 36 actually has *higher* usership than Chrome 40 (1.55% vs 1.47% of total market). This was how I justified to myself that killing all Chromes <41 was probably okay, given how we dropped <37 before.
